File: webApp/src/app/common/city.ts
/**
 * 城市实体
 */
export class City {
  /** id */
  id: number;

  /** 城市名 */
  name: string;

  pinyin: string;

  primaried: boolean;

  constructor(data?: { id?: number, name?: string, pinyin?: string, primaried?: boolean }) {
    if (data) {
      this.id = data.id;
      this.name = data.name;
      this.pinyin = data.pinyin;
      this.primaried = data.primaried;
    }
  }
}

File: bus-ticket/src/main/java/xiaoqiangZzz/busticket/Entity/City.java
package xiaoqiangZzz.busticket.Entity;


import com.mengyunzhi.core.entity.YunzhiEntity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class City implements YunzhiEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String pinyin; // 拼音

    private Boolean primaried = false; // 是否热门城市

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPinyin() {
        return pinyin;
    }

    public void setPinyin(String pinyin) {
        this.pinyin = pinyin;
    }

    public Boolean getPrimaried() {
        return primaried;
    }

    public void setPrimaried(Boolean primaried) {
        this.primaried = primaried;
    }
}
